### PR TITLE
Add Tests for Solver class

### PR DESCRIPTION
**Pull request by MrBrN197**
## Changes | ⚔
- ✔ Create class `Solver`
  - ☑ create tests for `Solver#factorial`
   - ☑ create tests for `Solver#reverse`
   - ☑ create tests for `Solver#fizzbuzz`

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  context 'test functionality of Solver methods' do
+    it 'should throw when number argument is < 0' do
+      expect { @solver.factorial(-1) }.to raise_error('argument cannot be less than zero', ArgumentError)
+    end
+
+    it 'test that factorial is calculated correctly' do
+      result = @solver.factorial(5)
+      expect(result).to eql 120
+    end
```
Comment (Lameck1) [inline spec/solver_spec.rb:21]: - [x] Well done team 👏🏼. We are missing one more test though. That is the special case i.e. when the input is `0`. Please test this edge case too. Let remind ourselves of what's quoted below:
  
  > - The factorial is the multiplication of all integers from 1 to N and has the **_`special case that the factorial of 0 is 1`_**

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @MrBrN197 & @wuyepabdul , 

So far so good 👍🏼. Let's take care of one more edge case in our test and we will be good to go.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (MrBrN197) [reply spec/solver_spec.rb:21]: Thanks

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
